### PR TITLE
Add tests to increase Output coverage.

### PR DESCRIPTION
**Pull request by DieBauer**
This merge request aims to increase the coverage of the `Output` class.

### CODE REVIEW

File: core/src/test/scala/io/finch/OutputSpec.scala [hunk 1]
```diff
@@ -12,6 +15,13 @@ class OutputSpec extends FinchSpec {
     check { o: Output[String] => o.toResponse[Text.Plain].status == o.status }
   }
 
+  it should "propagate overridden status to response" in {
+    check { (o: Output[String], s: Status) =>
+      val rep = o.withStatus(s).toResponse[Text.Plain]
+      rep.status === s
```
Comment (vkostyukov) [inline core/src/test/scala/io/finch/OutputSpec.scala]: Let's inline these into `o.withStatus(s).toResponse[Text.Plain].status === s`

File: core/src/test/scala/io/finch/OutputSpec.scala [hunk 2]
```diff
@@ -83,4 +93,56 @@ class OutputSpec extends FinchSpec {
         Encode[String, Text.Plain].apply(op.value, op.charset.getOrElse(StandardCharsets.UTF_8))
     }
   }
+
+  it should "create an empty endpoint with given status when calling unit" in {
+    check { s: Status =>
+      val rep = Output.unit(s).toResponse[Text.Plain]
```
Comment (vkostyukov) [inline core/src/test/scala/io/finch/OutputSpec.scala]: Please, inline `Output.unit(s).toResponse[Text.Plain].status === s`.

File: core/src/test/scala/io/finch/OutputSpec.scala [hunk 3]
```diff
@@ -83,4 +93,56 @@ class OutputSpec extends FinchSpec {
...
+      rep.status === s
+    }
+  }
+
+  it should "throw an exception on calling value on an Empty output" in {
+    import org.scalatest.prop.PropertyChecks
+    PropertyChecks.forAll { e: Output.Empty =>
+      the[IllegalStateException] thrownBy e.value should have message "empty output"
```
Comment (vkostyukov) [inline core/src/test/scala/io/finch/OutputSpec.scala]: Why using `forAll` directly here? Is that b/c of `the[Exception]` syntax? If we could do just wrap `e.value` with `Try` and test the equality instead.

File: core/src/test/scala/io/finch/OutputSpec.scala [hunk 4]
```diff
@@ -83,4 +93,56 @@ class OutputSpec extends FinchSpec {
...
+    }
+  }
+
+  it should "throw an exception on calling value on a Failure output" in {
+    import org.scalatest.prop.PropertyChecks
+    PropertyChecks.forAll { f: Output.Failure =>
```
Comment (vkostyukov) [inline core/src/test/scala/io/finch/OutputSpec.scala]: Same question about `forAll`.

File: core/src/test/scala/io/finch/OutputSpec.scala [hunk 5]
```diff
@@ -83,4 +93,56 @@ class OutputSpec extends FinchSpec {
...
+      the[Exception] thrownBy f.value should have message f.cause.getMessage
+    }
+  }
+
+  it should "check equivalences of arbitrary string outputs" in {
+    check {
+      forAll(genOutput[String], genOutput[String]) {
```
Comment (vkostyukov) [inline core/src/test/scala/io/finch/OutputSpec.scala]: Why we need both `check` and `forAll` here?

File: core/src/test/scala/io/finch/OutputSpec.scala [hunk 6]
```diff
@@ -83,4 +93,56 @@ class OutputSpec extends FinchSpec {
...
+  }
+
+  it should "check equivalences of failure outputs" in {
+    check { (fa: Output.Failure, fb: Output.Failure) =>
+      Output.outputEq[String].eqv(fa, fb) === (fa == fb)
+    }
+  }
+
+  it should "traverse arbitrary outputs" in {
+    check {
```
Comment (vkostyukov) [inline core/src/test/scala/io/finch/OutputSpec.scala]: Same question about `check` and `forAll`.

File: core/src/test/scala/io/finch/OutputSpec.scala (hunk 1 above)
Comment (DieBauer) [inline core/src/test/scala/io/finch/OutputSpec.scala]: Yes, will do.

File: core/src/test/scala/io/finch/OutputSpec.scala (hunk 3 above)
Comment (DieBauer) [inline core/src/test/scala/io/finch/OutputSpec.scala]: Good question, seems that check suffices then, will drop the scalatest syntactic sugar! thanks :)

**Comment by DieBauer** [comment]
@vkostyukov thanks for the review, I've applied changes as requested!

**Comment by vkostyukov** [comment]
Thank you again @DieBauer! This looks great and I especially like the +2% coverage increase!

**Pull request merged**